provisioning/service/src/Config/AttestationMechanism.cs

```diff
@@ -102,6 +107,17 @@ private AttestationMechanism(AttestationMechanismType type, TpmAttestation tpm,
                     break;
                 case AttestationMechanismType.None:
                     break;
+                case AttestationMechanismType.SymmetricKey:
+                    // In some cases symmetric keys are nulled out by the service
```

Comment:
Maybe I missed out when this was discussed - could you please explain this part?

Reply:
Good catch. This was copied from the internal repo, so @rajeevmv can you explain the case where the service gives/allows null symmetric keys?

Reply:
By default the service will not send back symmetric key since it contains the full key used by the device. This is different when compared to other type of attestations we support.

Reply:
The service will generate symmetric keys in the case when an Enrollment or an EnrollmentGroup does not have them during creation.

This is only applicable to the ProvisioningServiceClient.
Not to the ProvisioningDeviceClient.

---
In reply to: [247618950](https://github.com/Azure/azure-iot-sdk-csharp/pull/750#discussion_r247618950) [](ancestors = 247618950)